src/integrations/front-end-integration.php

```diff
@@ -306,23 +308,29 @@ public function query_loop_next_prev( $html, $block ) {
 	 * @return string The correct link.
 	 */
 	public function adjacent_rel_url( $link, $rel, $presentation = null ) {
```

Comment:
How about refactoring the dreaded multiple-conditions-nested-ifs into guard clauses? 😉 
```
	public function adjacent_rel_url( $link, $rel, $presentation = null ) {
		if ( $link ) {
			return $link;
		}

		if ( $rel !== 'next' && $rel !== 'prev' ) {
			return $link;
		}

		if ( \is_null( $this->$rel ) ) {
			return $link;
		}

		if ( ! \class_exists( WP_HTML_Tag_Processor::class ) ) {
			return $link;
		}

		$processor = new WP_HTML_Tag_Processor( $this->$rel );
		while ( $processor->next_tag( [ 'tag_name' => 'a' ] ) ) {
			$href = $processor->get_attribute( 'href' );

			if ( ! $href ) {
				continue;
			}

			// Safety check for full url, not expected.
			if ( \strpos( $href, 'http' ) === 0 ) {
				return $href;
			}

			// Check if $href is relative and append last part of the url to permalink.
			if ( \strpos( $href, '/' ) === 0 ) {
				$href_parts = \explode( '/', $href );
				return $presentation->permalink . \end( $href_parts );
			}
		}
		return $link;
	}
```